src/main/java/eu/europeana/metis/sandbox/service/workflow/OrderingService.java

```diff
@@ -12,5 +12,5 @@
    * @throws NullPointerException    if xml is null
    * @throws TransformationException if ordering fails
    */
-  String performOrdering(String xml) throws TransformationException;
+  String performOrdering(byte[] xml) throws TransformationException;
```

Comment:
Well is tricky because  `XsltTransformer.transform(byte[])` returns a `StringWriter` which I will probably still do a `toString().getBytes(Charset)` to get the byte[] as a return, which then will need to become a String again because validation library takes a String as a parameter.
But I am up to change all of this when libraries support receiving and returning a byte[] as the record representation, but I have the feeling that will take more work than it sounds

Reply:
[MET-2680](https://europeana.atlassian.net/browse/MET-2680)